File: sample-app.nest/src/interfaces/user.repository..ts
// import { User } from "src/models/user.entity";
import { User } from '@prisma/client';
export const USER_REPOSITORY = 'USER_REPOSITORY'; // Создаем строковый токен

export interface IUserRepository {
    getAll(): Promise<User[]>;
    create(user: Omit<User, 'id'>): Promise<User>;
    findAll(): Promise<User[]>;
    findOne(id: number): Promise<User>;
    update(id: number, user: Partial<User>): Promise<User>;
    remove(id: number): Promise<void>;
  }

File: kotlinapi/src/main/kotlin/com/example/kotlinapi/repositories/UserRepository.kt
package com.example.kotlinapi.repositories

import com.example.kotlinapi.models.User

interface UserRepository {
    fun save(user: User): User
    fun findById(id: Long): User?
    fun findAll(): List<User>
    fun deleteById(id: Long)
    fun update(user: User): User
}